cider/type/plan/BinaryExpr.cpp

```diff
@@ -86,41 +79,45 @@ JITExprValue& BinOper::codegen(JITFunction& func) {
   return expr_var_;
 }
 
-JITExprValue& BinOper::codegenFixedSizeColArithFun(JITValue& lhs, JITValue& rhs) {
+JITExprValue& BinOper::codegenFixedSizeColArithFun(JITValuePointer& null,
+                                                   JITValue& lhs,
+                                                   JITValue& rhs) {
   // TODO: Null Process
   switch (get_optype()) {
     case kMINUS:
-      return set_expr_value(nullptr, lhs - rhs);
+      return set_expr_value(null, lhs - rhs);
     case kPLUS:
-      return set_expr_value(nullptr, lhs + rhs);
+      return set_expr_value(null, lhs + rhs);
     case kMULTIPLY:
-      return set_expr_value(nullptr, lhs * rhs);
+      return set_expr_value(null, lhs * rhs);
     case kDIVIDE:
-      return set_expr_value(nullptr, lhs / rhs);
+      return set_expr_value(null, lhs / rhs);
     case kMODULO:
-      return set_expr_value(nullptr, lhs % rhs);
+      return set_expr_value(null, lhs % rhs);
     default:
       UNREACHABLE();
   }
 
   return expr_var_;
 }
 
-JITExprValue& BinOper::codegenFixedSizeColCmpFun(JITValue& lhs, JITValue& rhs) {
+JITExprValue& BinOper::codegenFixedSizeColCmpFun(JITValuePointer& null,
+                                                 JITValue& lhs,
+                                                 JITValue& rhs) {
   // TODO: Null Process
   switch (get_optype()) {
     case kEQ:
-      return set_expr_value(nullptr, lhs == rhs);
+      return set_expr_value(null, lhs == rhs);
```

Comment:
maybe we should refine this by adding a new API `set_expr_null` to avoid passing the null everywhere?

Reply:
maybe you should use `FixSizeJITExprValue`.

Reply:
It's just null value, should be passed whether it is nullptr or not.